configure.py

```diff
@@ -0,0 +1,413 @@
+    # variable, need to tell bazel to include PYTHONPATH
+    if environ_cp.get("PYTHONPATH"):
+        python_paths = environ_cp.get("PYTHONPATH").split(":")
+        if python_lib_path in python_paths:
+            write_action_env_to_bazelrc("PYTHONPATH", environ_cp.get("PYTHONPATH"))
+
+
+def set_cc_opt_flags(environ_cp):
+    """Set up architecture-dependent optimization flags.
+    Also append CC optimization flags to bazel.rc.
+    Args:
+      environ_cp: copy of the os.environ.
+    """
+    if is_ppc64le():
+        # gcc on ppc64le does not support -march, use mcpu instead
+        default_cc_opt_flags = "-mcpu=native"
+    elif is_windows():
+        default_cc_opt_flags = "/arch:AVX"
+    else:
+        # On all other platforms, no longer use `-march=native` as this can result
+        # in instructions that are too modern being generated. Users that want
+        # maximum performance should compile TF in their environment and can pass
+        # `-march=native` there.
+        # See https://github.com/tensorflow/tensorflow/issues/45744 and duplicates
+        default_cc_opt_flags = "-Wno-sign-compare"
+    question = (
+        "Please specify optimization flags to use during compilation when"
+        ' bazel option "--config=opt" is specified [Default is %s]: '
+    ) % default_cc_opt_flags
+    cc_opt_flags = get_from_env_or_user_or_default(
+        environ_cp, "CC_OPT_FLAGS", question, default_cc_opt_flags
+    )
+    for opt in cc_opt_flags.split():
+        write_to_bazelrc("build:opt --copt=%s" % opt)
+        write_to_bazelrc("build:opt --host_copt=%s" % opt)
+    write_to_bazelrc("build:opt --define with_default_optimizations=true")
+
+
+def maybe_set_manylinux_toolchain(environ_cp):
+    write_to_bazelrc(
+        "build:manylinux2010 --crosstool_top=//third_party/toolchains/preconfig/ubuntu16.04/gcc7_manylinux2010-nvcc-cuda10.1:toolchain"
+    )
+    if get_var(
+        environ_cp,
+        var_name="MANYLINUX2010",
+        query_item="manylinux2010-compatible pip package",
+        enabled_by_default=False,
+        question=(
+            "Are you trying to build a manylinux2010-compatible pip package "
+            "in the tensorflow:custom-op-ubuntu16 Docker container?"
+        ),
+        yes_reply="Building manylinux2010-compatible pip package.",
+        no_reply="Not building manylinux2010-compatible pip package.",
+    ):
+        write_to_bazelrc("build --config=manylinux2010")
+        write_to_bazelrc("test --config=manylinux2010")
+
+
+def set_windows_build_flags(environ_cp):
+    """Set Windows specific build options."""
+
+    # First available in VS 16.4. Speeds up Windows compile times by a lot. See
+    # https://groups.google.com/a/tensorflow.org/d/topic/build/SsW98Eo7l3o/discussion
+    # pylint: disable=line-too-long
+    write_to_bazelrc(
+        "build --copt=/d2ReducedOptimizeHugeFunctions --host_copt=/d2ReducedOptimizeHugeFunctions"
+    )
```

Comment:
Ah yes, I developed this PR in parallel with #590 which removes it from the `.bazelrc` (so moving it from there to the config file), which was simply because that's where TensorFlow has it. I think they probably did it this way because these flags are not supported prior to Visual Studio 2019 v16.4, and it's nicer to override the user bazelrc than the repo one? Idk though. Do we care about supporting Windows builds with older versions of Windows/older compilers?

Reply:
> Do we care about supporting Windows builds with older versions of Windows/older compilers?

I personally don't, especially considering the crazy build times with older compiler versions:
https://groups.google.com/u/1/a/tensorflow.org/g/build/c/SsW98Eo7l3o/discussion

TensorFlow enabled it by default in https://github.com/tensorflow/tensorflow/commit/9983bb3dee46da301b8294000984eb67fd2b995a (it was done conditionally before) so I think it would be save to move it to the global config. But feel free to keep it like this.

Reply:
Ah nice, yeah makes sense, in that case let's make it default, thanks :)